src/Pages/BookPage.js

```diff
@@ -40,7 +66,31 @@ function BookPage() {
 
     getEntry('Creator', illustratorId, setIllustrator);
   };
-  useEffect(getEntries, [bookId]); // Runs on mount and on change of bookId
+
+  // Gets list of recommended books
+  const getBooksLikeThis = async () => {
+    if (book) {
+      const recList = await RecFilter(
+        book.id,
+        book.fields.age_min,
+        book.fields.age_max,
+        book.fields.grade_min,
+        book.fields.grade_max,
+        book.fields['race/ethnicity'],
+        book.fields.genre,
+      );
+      console.log('recList: ', recList);
+    }
+  };
+
+  useEffect(() => {
+    pushToStorage();
+    getEntries();
+  }, [bookId]); // Runs on mount and on change of bookId
+
+  useEffect(() => {
```

Comment:
if you don't want to run the function on mount (only once book has loaded) you can use a ref to track what stage the application is at! If you have time we can go through this together, otherwise this looks good for now.